[3.7.0] Documentation for the Signature component (#1187)
telerik/blazor-docs: components/signature/overview.md

---
title: Overview
page_title: Signature Overview
description: Overview of the Signature component for Blazor.
slug: signature-overview
tags: telerik,blazor,signature,overview
published: True
position: 0
---

# Blazor Signature Overview

This article provides information about the <a href = "https://www.telerik.com/blazor-ui/signature" target="_blank">Blazor Signature component</a> and its core features.

> The Blazor Signature component provides an area where users can draw their signatures with the mouse, pen, or with their fingertips. You can use the component as part of any application where the users must sign a document. The Blazor Signature can be easily integrated with the [TelerikForm.]({%slug form-overview%}) The component stores the value of the signature as a <a href="https://learn.microsoft.com/en-us/dotnet/api/system.buffers.text.base64?view=net-6.0">base64 string</a>

Reviewer: ```suggestion
The Blazor Signature component provides an area where users can draw their signatures with a mouse, pen, or with their fingertips. You can use the component in any application where the users must sign a document. The Blazor Signature integrates easily with the [TelerikForm]({%slug form-overview%}). The component stores the value of the signature as a <a href="https://learn.microsoft.com/en-us/dotnet/api/system.buffers.text.base64?view=net-6.0">base64 string</a>
```

## Creating Blazor Signature

1. Use the `TelerikSignature` tag to add the component to your razor page.

> 1. Set the `Value` (one-way or two-way binding) parameter to a `string` property.

Reviewer: Breaking phrases with brackets affects readability (value parameter is a single "object" in the sentence). In general, brackets are not that common in English.

```suggestion
1. Set the `Value` parameter to a `string` property. It supports one-way and two-way binding.
```

> 1. Set the `Width` and `Height` parameters to `px` values.

Reviewer: Add a step for increasing the SignalR message size with a link to the below section. I immediately hit the default limit when I copied and ran the basic example locally.

>caption Blazor Signature with basic configuration

````CSHTML
<TelerikSignature @bind-Value="@SignatureValue" Width="300px" Height="300px">
</TelerikSignature>

@code {
    private string SignatureValue { get; set; }
}
````

## Appearance

> The Signature component provides settings to control its appearance. [Read more about the Blazor Signature appearance settings...]({%slug signature-appearance%})

Reviewer: Increase the paragraph value with some specific information. This will also hint the reader if they want to visit the linked page immediately:

```suggestion
The Signature component provides settings to control its appearance, for example colors and borders. [Read more about the Blazor Signature appearance settings...]({%slug signature-appearance%})
```

## Events

> The Blazor Signature **blur** and value **change** events to respond to user actions. [Read more about the Blazor Signature events...]({%slug signature-events%})

Reviewer: ```suggestion
Use the Blazor Signature **blur** and value **change** events to respond to user actions. [Read more about the Blazor Signature events...]({%slug signature-events%})
```

> ## Large Content Support

Reviewer: "Large" is likely unnecessary here. You can see that the max default limit is reached almost instantly with any signature that is more than 1-2 lines. Consider a heading like "Blazor Server Operation" or "SignalR Message Size" or whatever you like.

```
@SignatureValue.Length

<TelerikSignature @bind-Value="@SignatureValue" Width="300px" Height="300px" />

@code {
    private string SignatureValue { get; set; } = String.Empty;
}
```

>caption Program.cs (.NET 6)

<div class="skip-repl"></div>

````CS
using Microsoft.AspNetCore.SignalR;

var builder = WebApplication.CreateBuilder(args);

//...

// SignalR message size for the Signature
builder.Services.Configure<HubOptions>(options =>
{
    options.MaximumReceiveMessageSize = 1024 * 1024; // 1MB
});
````

>caption Startup.cs (.NET 5)

<div class="skip-repl"></div>

````CS
public void ConfigureServices(IServiceCollection services)
{
    //...

    // SignalR message size for the Signature
    services.Configure<HubOptions>(options =>
    {
        options.MaximumReceiveMessageSize = 1024 * 1024; // 1MB
    });
}
````

## Signature Parameters

The following table lists the Signature parameters. Also check the [Signature API Reference](/blazor-ui/api/Telerik.Blazor.Components.Signature).

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type and Default&nbsp;Value | Description |
| --- | --- | --- |

> | `AriaDescribedBy` | `string` | Maps to the `area-describedby` attribute. Use this parameter to establish a relationship between widgets or groups and the text that describes them. |

Reviewer: Start directly with what the parameter represents. By the way, what is "groups"?

```suggestion
| `AriaDescribedBy` | `string` | The `area-describedby` attribute. Use this parameter to establish a relationship between widgets or groups and the text that describes them. |
```

Author: Quite honestly, I have no idea, I did not understand what it does exactly, read that from the https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#:~:text=The%20aria%2Ddescribedby%20attribute%20lists,not%20limited%20to%20form%20controls.

Reviewer: In such cases, it may be better to just link the generic external docs and not deal with unclear content on our side.

| `AriaLabel` | `string` | Maps to the `aria-label` attribute. Use  this parameter if the text that labels the component is not visible. |
| `AriaLabelledBy` | `string` | Maps to the `area-labelledby` attribute. Use this parameter to reference another element to define its accessible name. |
| `Class` | `string` | Renders a custom CSS class to the `<div class="k-signature">` element. Use it to [override theme styles]({%slug themes-override%}). |
| `DebounceDelay` | `int` <br /> `(150)` | Time in milliseconds between the last typed symbol and the internal `oninput` event firing. |
| `Enabled` | `bool` <br /> `(true)` | Defines if the component is enabled. |

> | `ExportScale` | `double` | The `Width` and `Height` of the component will be multiplied by the value of the `ExportScale` when converting the signature to an image. |

Reviewer: ExportScale has no default value, but in this case, the component will use 2, so we may as well mention it.

> | `Height` | `string` | Defines the height of the component. Set the value of this parameter in `px`. |

Reviewer: ```suggestion
| `Height` | `string` | Defines the height of the component. Set it in `px`. |
```

| `HideLine` | `bool` | Whether the dotted line is rendered. |
| `Maximizable` | `bool` | Whether the Signature component can be maximized. |
| `PopupScale` | `double` | The `Width` and `Height` of the component will be multiplied by the value of the `PopupScale` when maximizing (`Maximizable="true"`) the Blazor Signature. |
| `ReadOnly` | `bool` | Maps to the `readonly` attribute. Use this parameter to put the Signature in read-only state. |
| `Smooth` | `bool` | Use this parameter to smoothen the signature line. |

> | `StrokeWidth` | `double?` | Set the width of the signature line. |

Reviewer: What is this signature line?

Author: when drawing a signature a line is rendered. Ideally, I would like to call it a stroke, but we can't explain a parameter by its name. What if people are not aware of what a stroke is?

I agree it's not optimal, but I could not come up with a better word.

Reviewer: I see. In this case consider "the drawn lines' **thickness** in pixels"
